capstone-ui/src/Components/InteractiveMap/InteractiveMap.jsx

```diff
@@ -0,0 +1,36 @@
+import React, { useEffect } from 'react';
+import * as topojson from 'topojson-client';
+import usData from '../../data/us.json';
+import "./InteractiveMap.css";
+
+export default function InteractiveMap() {
+    useEffect(() => {
+        const margin = { top: 50, left: 50, right: 50, bottom: 50 };
+        const height = 400 - margin.top - margin.bottom;
+        const width = 800 - margin.left - margin.right;
+
+        const svg = d3
+        .select('#map')
+        .append('svg')
+        .attr('height', height + margin.top + margin.bottom)
+        .attr('width', width + margin.left + margin.right)
+        .append('g')
+        .attr('transform', `translate(${margin.left}, ${margin.top})`);
+
+        const projection = d3.geoAlbersUsa().translate([width / 2, height / 2]).scale(800);
+
+        const path = d3.geoPath().projection(projection);
+
+        const states = topojson.feature(usData, usData.objects.states).features;
```

Comment:
I've seen they read in the json data in the resource video. Why is that not done here? I see it still works, but maybe we can try it that way to see if fixes your issue as well as callback code that should be executed after data is available. It also seems to allow you to do error handling.

Reply:
Since I have the json object directly, there's no real need to read and fetch the data. That would be use if I had to get things from JSON files and needed to wait before rendering the map or other components.

I did try it though, but the same issue persists.